public/hwclayer.h

```diff
@@ -348,7 +348,7 @@ struct HwcLayer {
       kVisible | kSurfaceDamageChanged | kVisibleRegionChanged | kZorderChanged;
   int layer_cache_ = kLayerAttributesChanged | kDisplayFrameRectChanged;
   bool is_cursor_layer_ = false;
-  uint32_t solid_color_ = 0;
+  uint32_t solid_color_ = 0xff;
```

Comment:
why and where solid color and alpha value can get messed?

Reply:
In glprogram.cpp fragment shader as below
    fragment_shader_stream << "  texSample = texture2D(uLayerTexture" << i
                           << ",\n"
                           << "                        fTexCoords[" << i
                           << "]);\n"
                           << "  texSample.rgb = texSample.rgb + uLayerColor[" << i
                           << "].rgb;\n"
                           << "  tempAlpha = min(texSample.a, uLayerColor[" << i
                           << "].a);\n"
                           << "  multRgb = texSample.rgb *\n"
                           << "            max(tempAlpha, uLayerPremult[" << i
                           << "]);\n"
                           << "  color += multRgb * uLayerAlpha[" << i
                           << "] * alphaCover;\n"
                           << "  alphaCover *= 1.0 - texSample.a * uLayerAlpha["
                           << i << "];\n";

tempAlpha = min(texSample.a, uLayerColor[i].a the alpha is the min of texture and layer.